### PR TITLE
Upgrade to upstream v0.15.0 and python3

### PR DESCRIPTION
**Pull request by milouse**
## Problem
- New upstream version 0.15.0
- Current package use python2 and upstream may remove python2 support one day or another (asciimoo/searx#1471)

## Solution
- Replace `virtualenv` by `virtualenv --python=/usr/bin/python3`

## PR Status
- [X] Code finished.
- [ ] Tested with Package_check (I'm not on debian/ubuntu :frowning_face:)
- [X] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [X] Can be reviewed and tested.

## Validation
---
*Minor decision*
- **Upgrade previous version** : 
- [ ] **Code review** : 
- [ ] **Approval (LGTM)** : 
- [ ] **Approval (LGTM)** : 
- **CI succeeded** : 
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/searx_ynh%20PR41%20(Official_fork)/) 

When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

**Comment by Josue-T** [comment]
The CI failled.

**Comment by anmol26s** [comment]
The upgrade from last version failed too. Will paste logs later today.

**Comment by milouse** [comment]
I add a new commit, trying to fix the upgrade problem. @anmol26s can you try again?

For the pathes problems, I must admit I'm not an expert of yunohost and don't know what happens. Any help will be appreciated ! Thanks

**Comment by anmol26s** [comment]
It is still failing.
Log
https://paste.yunohost.org/uxubexohad.go

**Comment by milouse** [comment]
Hi, your log is not complete. Could you copy/paste the content of the `/root/.pip/pip.log` file?

I'll try to do the test on my own. What is weird is that I succeeded to install it for me.

**Comment by maniackcrudelis** [comment]
I confirm that it doesn't work.
404 with Package_check and with a normal install.

**Comment by milouse** [comment]
I think I should rebase this PR on your branch @maniackcrudelis 
However I'll don't have time to do this before a while. Do not hesitate to take it over.

**Comment by nfk** [comment]
Hi,

Maybe it can help we have done the full move to python3 on [spot](https://gitlab.e.foundation/e/cloud/my-spot/commit/2cc4bf3eaa5cbab007b56ad4ac287d56fc57f5ae), Plus on spot we prefer the usage of docker instead of virtualenv. 

let me know if I can help.

File: scripts/install [hunk 1]
```diff
@@ -93,7 +93,7 @@ ynh_setup_source "$final_path"
 # INSTALL SEARX IN A VIRTUALENV
 #=================================================
 
-virtualenv --system-site-packages "$final_path"
+virtualenv --python=/usr/bin/python3 --system-site-packages "$final_path"
 set +u; source $final_path/bin/activate; set -u
 pip install -U setuptools
```
Comment (kay0u) [inline scripts/install]: Use `pip3` instead of `pip`

Comment (milouse) [reply scripts/install]: Oh yes. I'm not really used to debian defaults… You're right it totally make sense to do it that way.

Comment (milouse) [reply scripts/install]: Oh, sorry in fact you don't have to use pip3 (your comment was mis-aligned in the last window, thus I thought you were speaking about the virtualenv line). As we initiated virtualenv with python3, pip in that virtualenv is in fact pip3.

Comment (kay0u) [reply scripts/install]: I was not sure. Thank you for your explanation!

**Comment by kay0u** [comment]
Well, I tried to fix the issue, but I didn't succeed.
I think the problem is this one:

https://github.com/asciimoo/searx/issues/1337

If someone want to look at this...

**Comment by milouse** [comment]
I don't remember if I modified nginx config files :/ I'll try to give a second look to this MR ASAP. Thank you for your help!

**Comment by maniackcrudelis** [comment]
We're actually already in version 0.15, following that PR https://github.com/YunoHost-Apps/searx_ynh/pull/52.

**Comment by AkselRocks** [comment]
yunohost still uses python 2 and not python 3. so it's not a clean update.

**Comment by maniackcrudelis** [comment]
Feel free to modify this PR, or to open a new one to migrate to python 3

**Pull request closed without merging**